### PR TITLE
Refresh lookup data at periodic interval

### PR DESCRIPTION
**Pull request by tiziano88**
- Rename `load_lookup_entries` to `parse_lookup_entries`
- Put the `LookupData` instance behind an `Arc`
- Start a separate `tokio` task to refresh data periodically
- Add more logging at various points

Ref #1930

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_functions/loader/src/main.rs [hunk 1]
```diff
@@ -190,12 +215,18 @@ async fn main() -> anyhow::Result<()> {
...
-    let lookup_data = LookupData::new_empty(&config.lookup_data_url);
+    let lookup_data = Arc::new(LookupData::new_empty(&config.lookup_data_url));
     if !config.lookup_data_url.is_empty() {
-        lookup_data
-            .refresh()
-            .await
-            .context("could not load lookup data")?;
+        let lookup_data_download_period = match config.lookup_data_download_period {
+            Some(v) => v,
+            None => {
```
Comment (conradgrobler) [inline oak_functions/loader/src/main.rs:222]: Should we enforce this? I can imagine cases where the data is stable enough that refreshes might not be required outside of the normal restarts of the runtime.

File: oak_functions/loader/src/main.rs [hunk 2]
```diff
@@ -190,12 +215,18 @@ async fn main() -> anyhow::Result<()> {
     let wasm_module_bytes = fs::read(&opt.wasm_path)
         .with_context(|| format!("Couldn't read Wasm file {}", &opt.wasm_path))?;
 
-    let lookup_data = LookupData::new_empty(&config.lookup_data_url);
+    let lookup_data = Arc::new(LookupData::new_empty(&config.lookup_data_url));
     if !config.lookup_data_url.is_empty() {
-        lookup_data
```
Comment (conradgrobler) [inline oak_functions/loader/src/main.rs:195]: I think that this should either remain here, or the periodic refresh code should first do the refresh before waiting in the loop. At the moment it seems like the lookup data will remain empty for the first period if I am reading the logic correctly.

File: oak_functions/loader/src/main.rs (hunk 1 above)
Comment (tiziano88) [reply oak_functions/loader/src/main.rs]: Good point, made more lenient now.

File: oak_functions/loader/src/main.rs (hunk 2 above)
Comment (tiziano88) [reply oak_functions/loader/src/main.rs:191]: Done, I think it makes more sense now, thanks!

**Review by conradgrobler — APPROVED** [review]

**Pull request merged**